bindings/matrix-sdk-ffi/src/lib.rs

```diff
@@ -36,13 +36,15 @@ pub struct ClientState {
     has_first_synced: bool,
     is_syncing: bool,
     should_stop_syncing: bool,
+    is_soft_logout: bool,
 }
 
 #[derive(Serialize, Deserialize)]
 struct RestoreToken {
     is_guest: bool,
     homeurl: String,
     session: Session,
+    is_soft_logout: bool,
```

Comment:
let's give this a default so old tokens still serialize properly:
```suggestion
    #[serde(default)]
    is_soft_logout: bool,
```